File: declarative-android-hn/app/src/main/java/com/prof18/declarativeandroidhn/data/model/News.kt
package com.prof18.declarativeandroidhn.data.model

import java.text.SimpleDateFormat
import java.util.*

data class News(
    val by: String,
    val id: Long,
    val score: Int,
    val text: String? = null,
    val time: Long,
    val title: String,
    val type: String,
    val url: String? = null
) {

    fun getStringTime(): String {
        val formatter = SimpleDateFormat("d MMM yyyy", Locale.getDefault())
        return formatter.format(Date(time*1000))

    }

}

File: declarative_flutter_hn/lib/data/model/news.dart
import 'package:intl/intl.dart';

class News {
  String by;
  int id;
  int score;
  String text;
  int time;
  String title;
  String type;
  String url;

  String getStringTime() {
    var formatter = new DateFormat('d MMM yyyy');
    String formatted =
        formatter.format(DateTime.fromMillisecondsSinceEpoch(time * 1000));
    return formatted;
  }

  News(
      {this.by,
      this.id,
      this.score,
      this.text,
      this.time,
      this.title,
      this.type,
      this.url});
}
